CHANGELOG.md

```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+## [v2.4.0](https://github.com/allenai/OLMo-core/releases/tag/v2.4.0) - 2025-11-20
```

Comment:
Our release script will add a line like this automatically. So all we really need to do is update the version, which you did.

https://github.com/allenai/OLMo-core/blob/a25a5144440ff97488194951db9d5f8a6cc6ffc7/src/scripts/release/release.sh#L19